src/Qir/Execution/Tools/Executable/ClangClient.cs

```diff
@@ -23,13 +24,15 @@ public async Task CreateExecutableAsync(string[] inputFiles, string[] libraries,
 
             // string.Join does not automatically prepend the delimiter, so it is included again in the string here.
             var librariesArg = $"{LinkFlag} {string.Join(LinkFlag, libraries)}";
-            var arguments = $"{inputsArg} -I {includePath} -L {libraryPath} {librariesArg} -o {outputPath}";
+            var arguments = $"{inputsArg} -I {includePath} -L {libraryPath} {librariesArg} -o {outputPath} -std=c++17 -v";
             logger.LogInfo($"Invoking clang with the following arguments: {arguments}");
             var taskCompletionSource = new TaskCompletionSource<bool>();
             using var process = new Process();
+            Environment.SetEnvironmentVariable("DYLD_LIBRARY_PATH", libraryPath);
```

Comment:
I believe this is only relevant for execution, so we should remove this from here and maybe add it into the execution section. I am hesitant to set environment variables but we might not have a way around it.

Reply:
I think the argument can be made for having it in the clang client as it is rather specific to c++ execution and the very libraries that must be set in the environment variable are known at link-time.